monkey/agent_plugins/build_plugin.sh

```diff
@@ -50,27 +50,33 @@ plugin_options_filename="${plugin_name_lowercase}_options"
 plugin_options_filepath="${PLUGIN_PATH}/src/${plugin_options_filename}.py"
 plugin_options_model_name="${plugin_name}Options"
 
-python3.11 << EOF
+config_schema_generated=$(python3.11 << EOF
 
 import json
 from pathlib import Path
 
 def construct_config_schema():
   config_schema = {"type": "object"}
-  print("Generating config-schema.json from ${plugin_options_filepath}.")
   if Path("${plugin_options_filepath}").exists():
     from $plugin_options_filename import $plugin_options_model_name
     config_schema = {"properties": $plugin_options_model_name.model_json_schema()["properties"]}
 
-  with open("${PLUGIN_PATH}/config-schema.json", "w") as f:
+  with open("${PLUGIN_PATH}/${SCHEMA_FILENAME}", "w") as f:
     f.write(json.dumps(config_schema))
 
-if Path("${PLUGIN_PATH}/config-schema.json").exists():
-  print("\033[0m\033[91mSkipping generating config-schema. Reason: config_schema.json already exists \033[0m")
+if Path("${PLUGIN_PATH}/${SCHEMA_FILENAME}").exists():
+  print("1")
 else:
   construct_config_schema()
+  print("0")
 
 EOF
+)
+if [ "$config_schema_generated" -eq "1"  ]; then
```

Comment:
Use return code instead of stdout/stderr. Otherwise an unexpected error might cause your conditional to behave in unexpected ways.